### ISSUE TITLE
Admin JID not created when using MongoDB with Tigase

### ISSUE THREAD

**Issue by ShubjeetPal** [tigase/tigase-server#52]
When Installing Tigase with MongoDB the installation happens successfully without any issue. 
1) But the admin JID is not created automatically. 
2) Register option on Tigase Web Client also does not take BOSH/ Websocket url. 
3) REST API also requires authentication to create user.
This I have seen on Tigase 8.0.0, 8.1.0 on Windows 10 and on Linux server with JDK 11

Is this a bug? what is the way to create admin JID?

**Comment by woj-tek** [comment]
> But the admin JID is not created automatically.

Thank you for reporting this. I added it to our backlog (ref: mongodb-33)

> Register option on Tigase Web Client also does not take BOSH/ Websocket url.

This is a bug I'm afraid. You could try selecting "Other domain" and make sure it resolves to your destination.

Can you register a user using other XMPP client?

> REST API also requires authentication to create user.

This is correct and intentional.

**Comment by ShubjeetPal** [comment]
> Register option on Tigase Web Client also does not take BOSH/ Websocket url.
> This is a bug I'm afraid. You could try selecting "Other domain" and make sure it resolves to your destination.

Our Firewall is only allowing https (port 443) and http (80, 8080) traffic. We are trying to get the secure websocket (wss) port opened. All others ports are blocked. Since Register is not taking BOSH/ Websocket url, we are not able to create user.

> Can you register a user using other XMPP client?

PSI does not take BOSH/ Websocket url so I am trying through Pidgin for which I have raised another question #44 which I am following.

> REST API also requires authentication to create user.
This is correct and intentional.

Can API-Key (through config.tdsl) and Authorization be kept as separate mechanism to use REST API in these kind of scenarios.

So currently we are blocked, need one of the solution working so that we can move ahead.

**Comment by woj-tek** [comment]
> Our Firewall is only allowing https (port 443) and http (80, 8080) traffic. We are trying to get the secure websocket (wss) port opened. All others ports are blocked. Since Register is not taking BOSH/ Websocket url, we are not able to create user.

>> Can you register a user using other XMPP client?

> PSI does not take BOSH/ Websocket url so I am trying through Pidgin for which I have raised another question #44 which I am following.

You can change Tigase configuration of c2s component to listen on opened ports - see https://docs.tigase.net/tigase-server/8.1.0/Administration_Guide/html/#_connections

> Can API-Key (through config.tdsl) and Authorization be kept as separate mechanism to use REST API in these kind of scenarios.

Actually you should be able to access user-creation REST API with only API-KEY (which you should be able to create during installation).

**Comment by ShubjeetPal** [comment]
>Can API-Key (through config.tdsl) and Authorization be kept as separate mechanism to use REST API in these kind of scenarios.
Actually you should be able to access user-creation REST API with only API-KEY (which you should be able to create during installation).

In section 16.3.1 Rest Module it is mentioned 
```
In previous version it was possible to configure api-keys for REST module using entries within configuration file. In the recent version we decided to remove this configuration option. Now, by default Tigase XMPP Server requires API key to be passed to all requests and you need to configure them before you will be able to use REST API.
```
I tried with below but it was not accepting it
```
http {
        api-keys = [ 'test1234', 'test5678' ]
}
```

**Comment by woj-tek** [comment]
As stated in the documentation - it was possible *in the **previous*** versions. It's not possible in Tigase XMPP Server 8.1.0 and newer. Currently the only option (without having admin user created by the installer) is to provide REST API key in the installer.

**Comment by ShubjeetPal** [comment]
>Currently the only option (without having admin user created by the installer) is to provide REST API key in the installer.

The web installer (http://localhost:8080/setup/) does not have any option to enter API key. Can you clarify how in the web installer I can view this option during installation.

**Comment by woj-tek** [comment]
It should have after the database is installed in step "HTTP REST - security configuration", see [6.3.5. Step Through the Installation Process](https://docs.tigase.net/tigase-server/master-snapshot/Administration_Guide/html/#_step_through_the_installation_process) 

![](https://docs.tigase.net/tigase-server/master-snapshot/Administration_Guide/html/images/admin/web-install-08.png)

**Comment by ShubjeetPal** [comment]
>It should have after the database is installed in step "HTTP REST - security configuration", see 6.3.5. Step Through the Installation Process

The web installer screens have changed and is not update in the Administration Guide. This option to set HTTP REST configuration does not appear any more (see the options on the left). This screen should be added back and the document updated

![image](https://user-images.githubusercontent.com/67587907/93000552-dea39600-f546-11ea-9d94-b1b9d88a93f1.png)

**Comment by woj-tek** [comment]
Apologies, yes - installer was simplified.

Regarding issue with adding admin user with mongodb selected - it will be fixed in the future.

**Comment by ShubjeetPal** [comment]
> 1. The admin JID is not created automatically during installation with MongoDB.
Thank you for reporting this. I added it to our backlog (ref: mongodb-33)

Can you please share the link to view/track the status of this item

>2. Register option on Tigase Web Client also does not take BOSH/ Websocket url.
This is a bug I'm afraid. You could try selecting "Other domain" and make sure it resolves to your destination.

Can you please share the link to view/ track the status of this bug

> 3. Web Installer was simplified and API Key screen removed from during initial setup (section 6.3 not updated). So API Key can not be added without having admin JID created. 

Is this is the intended behavior of API Key? 
And don't you think the document should be updated correctly as it is misleading for users and you also made a mistake. Is there any plan to update/clean the whole Administration Guide document so that different sections gives the correct details.

**Comment by woj-tek** [comment]
>>> The admin JID is not created automatically during installation with MongoDB.
>> Thank you for reporting this. I added it to our backlog (ref: mongodb-33)
> Can you please share the link to view/track the status of this item

https://projects.tigase.net/issue/mongodb-33

>>> Register option on Tigase Web Client also does not take BOSH/ Websocket url.
>> This is a bug I'm afraid. You could try selecting "Other domain" and make sure it resolves to your destination.
> Can you please share the link to view/ track the status of this bug

Current SureIM client is considered legacy and is in maintenance mode. This issue is considered non-critical and won't be fixed. We are working on new web client based on our new library written in Kotlin (https://github.com/tigase/halcyon/).

> And don't you think the document should be updated correctly as it is misleading for users and you also made a mistake.
> Is there any plan to update/clean the whole Administration Guide document so that different sections gives the correct details.

We strive to maintain the documentation on pair with the code but we have limited resources. Considering Tigase XMPP Server is **FREE** and **open source** we do accept Pull Requests.

**Comment by tanzhenchao** [comment]
I just tested tigase-server-8.2.0 and the problem still exists.
Can you provide a manual fix method?

**Comment by woj-tek** [comment]
> Can you provide a manual fix method?

You can just register the account using regular registration from the XMPP client.

**Comment by tanzhenchao** [comment]
> > Can you provide a manual fix method?
> 
> You can just register the account using regular registration from the XMPP client.
Thank you very much, we have completed the registration according to your prompts, using the following tools,
http://localhost:8080/ui

**Comment by woj-tek** [comment]
It will be fixed in version 8.2.0
